src/NzbDrone.Core/Notifications/Pushsafer/PushsaferSettings.cs

```diff
@@ -0,0 +1,39 @@
+﻿using FluentValidation;
+using NzbDrone.Core.Annotations;
+using NzbDrone.Core.ThingiProvider;
+using NzbDrone.Core.Validation;
+
+namespace NzbDrone.Core.Notifications.Pushsafer
+{
+    public class PushsaferSettingsValidator : AbstractValidator<PushsaferSettings>
+    {
+        public PushsaferSettingsValidator()
+        {
+            RuleFor(c => c.ApiKey).NotEmpty();
+        }
+    }
+
+    public class PushsaferSettings : IProviderConfig
+    {
+        private static readonly PushsaferSettingsValidator Validator = new PushsaferSettingsValidator();
+
+        [FieldDefinition(0, Label = "API Key", HelpLink = "https://www.pushsafer.com/dashboard")]
+        public string ApiKey { get; set; }
+
+        [FieldDefinition(1, Label = "Sound", Type = FieldType.Textbox, HelpText = "Notification sound, leave blank to use the default, set 0 for silent, set 1-50 for custom", HelpLink = "https://www.pushsafer.com/en/pushapi")]
+        public string Sound { get; set; }
+
+        [FieldDefinition(2, Label = "Icon", Type = FieldType.Textbox, HelpText = "Notification icon, leave blank to use the default, set 1-176 for custom", HelpLink = "https://www.pushsafer.com/en/pushapi")]
+        public string Icon { get; set; }
+
+        [FieldDefinition(3, Label = "Device", Type = FieldType.Textbox, HelpText = "Leave blank to notify all devices, set Single Device ID or Device Group ID for custom devices", HelpLink = "https://www.pushsafer.com/en/pushapi")]
+        public string Device { get; set; }
+
+        public bool IsValid => !string.IsNullOrWhiteSpace(ApiKey);
```

Comment:
What is this for?

Reply:
The API allows small groups of devices to be individually notified, Or all devices can be notified.

Reply:
I meant the IsValid property, not the Device.

Reply:
Nothing actually I am removing it. :)